src/nvim/api/ui.c

```diff
@@ -405,6 +405,8 @@ static void remote_ui_highlight_set(UI *ui, int id)
 static void remote_ui_grid_cursor_goto(UI *ui, Integer grid, Integer row,
                                        Integer col)
 {
+  grid = ui->ui_ext[kUIMultigrid] ? grid : DEFAULT_GRID_HANDLE;
```

Comment:
It is better to do this in `ui.c`, where the offset is added (so the reference to the original grid doesn't make any sense anymore)